pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
Ahh... got it. (Sorry I'm slow :-) ) Where do we do that? Do we do it in conversion or as a defaulting function for the dc? 

And why would we want to do that? Isn't it enough to explicitly set it in new-app ?